core/src/main/java/fr/pilato/elasticsearch/crawler/fs/FsParserAbstract.java

```diff
@@ -469,6 +469,9 @@ private void indexFile(FileAbstractModel fileAbstractModel, ScanStatistic stats,
                 } else if (fsSettings.getFs().isXmlSupport()) {
                     // https://github.com/dadoonet/fscrawler/issues/185 : Support Xml files
                     doc.setObject(XmlDocParser.generateMap(inputStream));
+                } else if (fsSettings.getFs().isSkipTika()) {
```

Comment:
I was thinking of it today and adding some tests. I found out that we should move this logic in `TikaDocParser#generate`  method.

Two reasons:

* We can more easily unit test this
* `UploadAPI` class (Rest API) must also see this setting. UploadAPI calls `generate` method. That's why we should call the logic from there.

Could you change that please?